src/pudl/etl/epacems_assets.py

```diff
@@ -49,106 +43,68 @@ def _partitioned_path() -> Path:
-    }
-    for year in years:
-        yield DynamicOutput(year, mapping_key=str(year))
+    for year_quarter in epacems_settings.year_quarters:
+        yield DynamicOutput(year_quarter, mapping_key=str(year_quarter))
 
 
 @op(
     required_resource_keys={"datastore", "dataset_settings"},
     tags={"memory-use": "high"},
 )
-def process_single_year(
+def extract_quarter(
     context,
-    year,
-    core_epa__assn_eia_epacamd: pd.DataFrame,
-    core_eia__entity_plants: pd.DataFrame,
-) -> YearPartitions:
-    """Process a single year of EPA CEMS data.
+    year_quarter: str,
+) -> str:
+    """Extract a single quarter of EPA CEMS data and write to parquet.
 
     Args:
         context: dagster keyword that provides access to resources and config.
-        year: Year of data to process.
-        core_epa__assn_eia_epacamd: The EPA EIA crosswalk table used for harmonizing the
-            ORISPL code with EIA.
-        core_eia__entity_plants: The EIA Plant entities used for aligning timezones.
+        year_quarter: Year quarter to process, formatted like '1995q1'.
     """
     ds = context.resources.datastore
-    epacems_settings = context.resources.dataset_settings.epacems
 
-    schema = Resource.from_id("core_epacems__hourly_emissions").to_pyarrow()
     partitioned_path = _partitioned_path()
+    partition = EpaCemsPartition(year_quarter=year_quarter)
+
+    ds = EpaCemsDatastore(context.resources.datastore)
+    (
+        ds.get_data_frame(partition=partition)
+        .with_columns(year=partition.year)
+        .sink_parquet(partitioned_path / f"{year_quarter}.parquet")
+    )
+
+    return year_quarter
+
 
-    year_quarters_in_year = {
-        yq
-        for yq in epacems_settings.year_quarters
-        if EpaCemsPartition(year_quarter=yq).year == year
-    }
-
-    for year_quarter in year_quarters_in_year:
-        logger.info(f"Processing EPA CEMS hourly data for {year_quarter}")
-        df = pudl.extract.epacems.extract(year_quarter=year_quarter, ds=ds)
-        if not df.empty:  # If state-year combination has data
-            df = pudl.transform.epacems.transform(
-                df, core_epa__assn_eia_epacamd, core_eia__entity_plants
-            )
-        table = pa.Table.from_pandas(df, schema=schema, preserve_index=False)
-
-        # Write to a directory of partitioned parquet files
-        with pq.ParquetWriter(
-            where=partitioned_path / f"epacems-{year_quarter}.parquet",
-            schema=schema,
-            compression="snappy",
-            version="2.6",
-        ) as partitioned_writer:
-            partitioned_writer.write_table(table)
-
-    return YearPartitions(year_quarters_in_year)
-
-
-@op
-def consolidate_partitions(context, partitions: list[YearPartitions]) -> None:
-    """Read partitions into memory and write to a single monolithic output.
+@op(out={"core_epacems__hourly_emissions": Out(io_manager_key="parquet_io_manager")})
```

Comment:
This is now using the `parquet_io_manager` as a proof of concept. The downstream asset is also using the io-manager with Polars.